### PR TITLE
added custom matchers for lwrps; added spec coverage for default recipe

### PR DESCRIPTION
**Pull request by xacaxulu**

### CODE REVIEW

File: providers/instance.rb [hunk 1]
```diff
@@ -3,19 +3,12 @@
...
-      # not_if { ::File.directory?("/opt/vmware/vfabric-tc-server-standard/#{new_resource.instance_name}") }
-      creates "/opt/vmware/vfabric-tc-server-standard/#{new_resource.instance_name}"
-    end
+    command "./tcruntime-instance.sh create #{new_resource.instance_name} --force --property base.runtime.user=root"
   end
 
   link "/etc/init.d/tcserver-#{new_resource.instance_name}" do
     owner 'root'
     group 'root'
-    link_type :symbolic
```
Comment (xacaxulu) [inline providers/instance.rb]: default is :symbolic

**Comment by charlesjohnson** [comment]
Generally looks good. A couple notes:

- Can you leave out the changes to kitchen.yml and gitignore the role?
- I don't remember anymore why I made 'force' a tunable attribute to the lwrp. What are the cases where I wouldn't want to force it?
- If we remove 'force' from the provider we should probably also yank it from the resource.
- Would it make more sense for the logic that checks for a "running as" regex live inside the tcserver_instance lwrp as a guard?
- Is there a better tcserver rpm than dropbox, or is it all tied up behind vmware clickwalls?

**Comment by xacaxulu** [comment]
* Can you leave out the changes to kitchen.yml and gitignore the role?
  Done
* I don't remember anymore why I made 'force' a tunable attribute to the lwrp. What are the cases where I wouldn't want to force it?
  Since we now check for a running process, we don't need to force install. We could always add in a `restart`.
* If we remove 'force' from the provider we should probably also yank it from the resource.
  Done.
* Would it make more sense for the logic that checks for a "running as" regex live inside the tcserver_instance lwrp as a guard?
  Perhaps. I just like the not_if block to be visible to the coder as they usually look at recipes/default.rb first. I'm not wed to it either way.

* It appears that tcserver is now served by Pivotal, and is locked behind a loginwall here: https://network.pivotal.io/products/pivotal-tcserver#/releases/3-1-0 We should remove all links to the actual RPM, regardless of where it's hosted. Users need to download it themselves and point to their own copy.
  This has been my experience. Most companies have their custom version in source control behind their own firewall. Done

**Comment by charlesjohnson** [comment]
Excellent, thank you!

**Pull request merged**